File: javascript/717.1-bit-and-2-bit-Characters.js
/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function (bits) {
  let i = 0;

  while (i <= bits.length - 2) {
    if (bits[i] === 1) {
      i += 2;
    } else {
      i += 1;
    }
  }

  return i + 1 === bits.length;
};

File: python/717.1-bit-and-2-bit-Characters.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        i = 0

        while i <= len(bits) - 2:
            if bits[i] == 1:
                i += 2
            else:
                i += 1

        return i + 1 == len(bits)